apps/browser/src/vault/popup/components/vault-v2/item-more-options/item-more-options.component.ts

```diff
@@ -103,6 +105,23 @@ export class ItemMoreOptionsComponent {
     }),
   );
 
+  /** Observable Boolean checking if item can show Archive menu option */
+  protected canArchive$ = combineLatest([
+    this._cipher$,
+    this.accountService.activeAccount$.pipe(getUserId),
+  ]).pipe(
+    filter(([cipher, userId]) => cipher != null && userId != null),
+    switchMap(([cipher, userId]) => {
+      return this.cipherArchiveService.userCanArchive$(userId).pipe(
+        map((canArchive) => {
+          return (
+            canArchive && !CipherViewLikeUtils.isArchived(cipher) && cipher.organizationId == null
+          );
+        }),
+      );
+    }),
+  );
```

Comment:
We can reduce nesting here to make it a little easier to understand:
```suggestion 
protected canArchive$ = combineLatest([
    this._cipher$,
    this.accountService.activeAccount$.pipe(
      getUserId,
      switchMap((userId) => this.cipherArchiveService.userCanArchive$(userId)),
    ),
  ]).pipe(
    filter(([cipher, userId]) => cipher != null && userId != null),
    map(
      ([cipher, canArchive]) =>
        canArchive && !CipherViewLikeUtils.isArchived(cipher) && cipher.organizationId == null,
    ),
  )
```